docs/src/pages/guides/typescript.md

```diff
@@ -92,7 +94,38 @@ const DecoratedNoProps = decorate<{}>( // <-- note the type argument!
 );
 ```
 
-To avoid worrying about this edge case it may be a good habit to always provide an explicit type argument to `decorate`.
+Scenario 2: `Props` is a union type. To avoid getting a compiler error, you'll need to provide an explict type argument: 
```

Comment:
"Again, to avoid getting a compiler error..."

Reply:
yep, I agree that's clearer. changed